core-processor/src/ext.rs

```diff
@@ -483,31 +502,14 @@ impl EnvExt for Ext {
     }
 
     fn charge_gas(&mut self, val: u64) -> Result<(), Self::Error> {
-        use ChargeResult::*;
-
         let common_charge = self.context.gas_counter.charge(val);
         let allowance_charge = self.context.gas_allowance_counter.charge(val);
-
-        let res: Result<(), ProcessorError> = match (common_charge, allowance_charge) {
-            (NotEnough, _) => Err(ExecutionError::GasLimitExceeded.into()),
-            (Enough, NotEnough) => Err(TerminationReason::GasAllowanceExceeded.into()),
-            (Enough, Enough) => Ok(()),
-        };
-
-        self.return_and_store_err(res)
+        self.check_charge_results(common_charge, allowance_charge)
```

Comment:
That's not intuitive for me that `check_charge_results` uses `return_and_store_err()`

Reply:
So I'd split it into separate calls

Reply:
Methods above have the same behavior